### PR TITLE
Update to v2.0 of Appendix D & other updates

### PR DESCRIPTION
**Pull request by smartbit**
Update to match [Appendix D 2.0.pdf](https://github.com/AnwarYagoub/RHCSA-RHCE-Lab-Environment/files/459076/Appendix.D.2.0.pdf) and be consistent in the labs eg. 
- `192.168.4.0/24` instead of `192.168.122.0/24`
- only use primary DNS `8.8.8.8`, removed secondary `8.8.4.4` (although this is not a best practice)

Other items
- added `haveged` to [speedup](http://blog-ftweedal.rhcloud.com/2014/05/more-entropy-with-haveged/) install of FreeIPA
- added `bash-completion` to assist with those pesky `nmcli`commands
-  install [EPEL](https://fedoraproject.org/wiki/EPEL) and update cache-proxy with [epel-mirrors](https://mirrors.fedoraproject.org/mirrorlist?repo=epel-7&arch=x86_64&country=global) and [allow https](https://www.pitt-pladdy.com/blog/_20150720-132951_0100_Home_Lab_Project_apt-cacher-ng_with_CentOS/)
- use _dynamic_ [CentOS mirrors](https://www.centos.org/download/full-mirrorlist.csv) list
- lower case `freeipa` for consistency on vagrant command line

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -20,14 +21,15 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     end
   end
 
+  config.ssh.insert_key = false # for vagrant 1.8.5 see https://github.com/mitchellh/vagrant/issues/7610#issuecomment-234609660
```
Comment (AnwarYagoub) [inline Vagrantfile]: @smartbit i faced this issue yesterday & i fix it by applying a patch  from this link https://github.com/mitchellh/vagrant/issues/7610

**Comment by AnwarYagoub** [comment]
Hello @smartbit & thanks for contributing in this project. i will be more than happy to accept this PR but why removing the secondary the secondary DNS.

**Comment by smartbit** [comment]
Hi Anwar, I haven't done all exercises so I can't say what the impact might be other that the exercises will be different. Appendix D v2.0 mentions setting _DNS temporarily to 8.8.8.8_ and at 19) to remove it again and point to the internal DNS 192.168.4.200. Adding secondary DNS adds confusion and I try to follow the KISS principle.

Now we talk about it, I think the two ethernet controllers might be confusing too. Maybe we can come up with a vagrant setup with 192.168.4.0/24 and NATing at the same time. With manual installing CentOS this is possible. Sander is using [VMware Fusion](https://www.vmware.com/support/pubs/fusion_pubs.html), which has different network topology than VirtualBox.

File: Vagrantfile (hunk 1 above)
Comment (smartbit) [reply Vagrantfile]: You mean /opt/vagrant/embedded/gems/gems/vagrant-1.8.5/plugins/guests/linux/cap/public_key.rb ? How to execute it?

Comment (AnwarYagoub) [reply Vagrantfile]: @smartbit 
the problem is **/home/vagrant/.ssh/authorized_keys** permission within the guest.

**Solution 1:**
manually insert the following line to ( /opt/vagrant/embedded/gems/gems/vagrant-1.8.5/plugins/guests/linux/cap/public_key.rb) after line 56 (mv ~/.ssh/authorized_keys.tmp ~/.ssh/authorized_keys):

chmod 0600 ~/.ssh/authorized_keys

**Solution 2:**
create patch file with the following contents

``` shell
56a57
>               chmod 0600 ~/.ssh/authorized_keys
```

and apply the patch

``` shell
$ sudo patch /opt/vagrant/embedded/gems/gems/vagrant-1.8.5/plugins/guests/linux/cap/public_key.rb < patch_file
```

**Comment by AnwarYagoub** [comment]
I have added the secondary if someone wanted to use other DNS servers than the one used in this exercises, but removing the secondary  DNS will match the environment with the description from Appendix D so no problem removing it.

As for network controllers Interface **eth0** in NAT interface while interface **eth1** in used for 192.168.4.0/24 network. is it possible to do both using the same controller? i don't know.

**Pull request merged**